packages/sc-deployer/src/index.ts

```diff
@@ -224,28 +302,26 @@ async function deploySC(
 
   console.log('Waiting for events...');
 
-  // await finalization if required
-  await awaitOperationFinalization(client, opId);
+  // async poll events in the background for the given opId
+  const { isError, eventPoller, events }: IEventPollerResult =
+    await time.withTimeoutRejection<IEventPollerResult>(
```

Comment:
Not directly related to this PR, but its a shame that we reimplemented theses time related function.. everything already exists somewhere in an npm package and we should not reinvent the wheel. this "time" object has nothing to do in our web3 repo...
wait function: https://www.npmjs.com/package/delay
set Timeout: https://www.npmjs.com/package/await-timeout
etc...

Reply:
Do you insist on using the delay and await-timeout packages ? Can I do that in a separate MR ?

Reply:
he approve so you should create an issue to to in later